src/sagemaker/jumpstart/curated_hub/types.py

```diff
@@ -19,6 +19,16 @@
 
 from sagemaker.jumpstart.types import JumpStartDataHolderType, JumpStartModelSpecs
 
+class CuratedHubTagName(str, Enum):
+    """Enum class for Curated Hub """
+    DEPRECATED_VERSIONS_TAG = "deprecated_versions"
+    TRAINING_VULNERABLE_VERSIONS_TAG = "training_vulnerable_versions"
+    INFERENCE_VULNERABLE_VERSIONS_TAG = "inference_vulnerable_versions"
+
+@dataclass
+class Tag:
```

Comment:
I'd argue we don't need this and can use that typing from the SDK. The generic naming is confusing, and it doesn't deserve it's own `CuratedHubUnsupportedTag` class imo

Reply:
This is to help tamp down the `key` values to specific `CuratedHubTagName` for typechecking at compile time. As this is customer-facing, I'd argue that making sure that we don't accidentally pass in `str` that aren't reviewed by the team